tests/test_algorithms.py

```diff
@@ -0,0 +1,36 @@
+import aesara
+import aesara.tensor as at
+import pytest
+
+from aehmc import algorithms
+
+
+def test_dual_averaging():
+    """Find the minimum of a simple function using Dual Averaging."""
+
+    def fn(x):
+        return (x - 1) ** 2
+
+    init, update = algorithms.dual_averaging(gamma=0.5)
+
+    def one_step(step, x, x_avg, gradient_avg):
+        value = fn(x)
+        gradient = aesara.grad(value, x)
+        return update(gradient, step, x, x_avg, gradient_avg)
+
+    x_init = at.as_tensor(0.0, dtype="float64")
```

Comment:
Does this fail if you use `aesara.config.floatX`?

Reply:
Try it with [a `conftest.py` setup like Aesara's](https://github.com/aesara-devs/aesara/blob/bda02f3c0513cba933f92cc293fdf129e5ff24a2/conftest.py#L10).  That will make sure the setting is available _before_ the imports.

Reply:
I added the `conftest.py` file with `floatX` set to `float32`; `print(aesara.config.floatX)` in the `test_dual_averaging` function return `float32`. However the test still fails with the following error:

```
ValueError: When compiling the inner function of scan the following error has been encountered: The initial state (`outputs_info` in scan nomenclature) of variable IncSubtensor{Set;:int64:}.0 (argument number 1) has dtype float32, while the result of the inner function (`fn`) has dtype float64. This can happen if the inner function of scan results in an upcast or downcast.
```

Reply:
I found [this](https://stackoverflow.com/questions/35518155/why-does-scan-upcast) related question with a simpler example on SO.

Reply:
I just provided an answer to that question.